Sources/Moya/Task+Internal.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+import Moya
+
+extension Task {
+
+    /// Returns all `TaskParametersType` objects to use when generating the `URLRequest`.
+    func allParameters() -> [TaskParametersType] {
+        var providers: [TaskParametersType?] = [bodyParams, urlParams]
```

Comment:
Nit: Could this be done like
```swift
var providers: [TaskParametersType?] = [bodyParams, urlParams, customParams]
return providers.compactMap { $0 }

// Or even
return  [bodyParams, urlParams, customParams].compactMap { $0 }
```
?

Reply:
@jdisho [had the same reaction](https://github.com/Moya/Moya/pull/1936#discussion_r345861114) ^^  `customParams` is an array while `bodyParams` and `urlParams` are not.